File: CamelCase Pattern Matching - GFG/camel-case-pattern-matching.java
//{ Driver Code Starts
//Initial Template for Java

import java.io.*;
import java.util.*;

class GFG {
    public static void main(String args[]) throws IOException {
        BufferedReader read =
            new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(read.readLine());
        while (t-- > 0) {
            int N = Integer.parseInt(read.readLine());
            String[] Dictionary=read.readLine().split(" ");
            String Pattern=read.readLine();
            Solution ob = new Solution();
            ArrayList <String> ans=ob.CamelCase(N,Dictionary,Pattern);
            for(String u:ans)
             System.out.print(u+" ");
            System.out.println(); 
        }
    }
}
// } Driver Code Ends


//User function Template for Java

class Solution{
    ArrayList<String> CamelCase(int N,String[] Dictionary,String Pattern){
    ArrayList<String> fn = new ArrayList<>();
        for (String x : Dictionary) {
            String ans = "";
            for (char i : x.toCharArray()) {
                if (Character.isUpperCase(i)) {
                    ans += i;
                }
                if (Pattern.equals(ans.substring(0, Math.min(Pattern.length(), ans.length())))) {
                    fn.add(x);
                    break;
                }
            }
        }
    if(fn.size() == 0){
        fn.add("-1");
        return fn;
    }
    Collections.sort(fn);
    return fn;
    }
}

File: CamelCase Pattern Matching - GFG/camel-case-pattern-matching.py
#User function Template for python3

class Solution:
    def CamelCase(self,N,Dictionary,Pattern):
        
        fn = []
        for x in Dictionary:
            ans = ""
            for i in x:
                if i.isupper() == True:
                    ans += i
                if Pattern == ans[:len(Pattern)]:
                    fn.append(x)
                    break
        if fn:
            return fn
        else:
            return [-1]


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__=='__main__':
    t=int(input())
    for _ in range(t):
        N=int(input())
        Dictionary=list(map(str,input().split()))
        Pattern=input()
        ob=Solution()
        ans=ob.CamelCase(N,Dictionary,Pattern)
        ans.sort()
        for i in ans:
            print(i,end=" ")
        print()    
# } Driver Code Ends